cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -123,6 +134,16 @@ public AppFabricServer(CConfiguration configuration, DiscoveryService discoveryS
     this.pluginService = pluginService;
     this.privilegesFetcherProxyService = privilegesFetcherProxyService;
     this.routeStore = routeStore;
+    this.sslEnabled = configuration.getBoolean(Constants.Security.AppFabric.SSL_ENABLED);
+    if (isSSLEnabled()) {
+      this.serverPort = configuration.getInt(Constants.AppFabric.SERVER_SSL_PORT);
+      KeyStore ks = GeneratedCertKeyStoreCreator.getSSLKeyStore(sConf);
+      String password = sConf.get(Constants.Security.AppFabric.SSL_KEYSTORE_PASSWORD);
```

Comment:
It is better to do the validations here and fail, rather than throwing RuntimeExceptions below.